src/views/MyPage/components/DesignerInfoSection.tsx

```diff
@@ -1,132 +1,223 @@
 import styled from 'styled-components';
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 
 import ProfileUpload from '@/views/@common/components/ProfileUpload';
 import TitleField from '@/views/@common/components/TitleField';
 import TextArea200 from '@/views/@common/components/TextArea200';
 import Input from '@/views/@common/components/Input';
+import PostCode from '@/views/@common/components/PostCode';
 import { IcInformation } from '@/views/@common/assets/icons';
 import { IcSearch } from '@/views/SignUpPage/assets/icons';
 import { MESSAGE } from '../constants/message';
 import { DAYS } from '@/views/@common/constants/days';
-
-interface ProfileImgInfoProps {
-  imgUrl: string;
-  imgObj: File;
-}
+import { DUMMY_DATA } from '../constants/dummy';
+import Header from '@/views/@common/components/Header';
 
 interface DesignerInfoSectionProps {
   onInfoChange: () => void;
 }
 
 const DesignerInfoSection = ({ onInfoChange }: DesignerInfoSectionProps) => {
-  const [isClicked] = useState<string[]>(['', '', '', '', '', '']);
+  const { profileImg, introduction, designerInfo } = DUMMY_DATA.data;
+  const [AddressModal, setAddressModal] = useState(false);
+
+  const [info, setInfo] = useState({
+    profileImg: profileImg,
+    introduction: introduction,
+    name: designerInfo.name,
+    gender: designerInfo.gender,
+    shopName: designerInfo.hairShop.name,
+    Address: designerInfo.hairShop.address,
+    dayOff: Object.keys(DAYS).map((day) => (designerInfo.dayOffs.includes(DAYS[day]) ? DAYS[day] : '')),
+    DetailAddress: designerInfo.hairShop.detailAddress,
+    instagramUrl: designerInfo.portfolio.instagramUrl,
+    naverPlaceUrl: designerInfo.portfolio.naverPlaceUrl,
+    OpenChatUrl: designerInfo.kakaoOpenChatUrl,
+  });
+
+  const handleDayOffClick = (event: React.MouseEvent<HTMLButtonElement>, index: number) => {
+    const dayValue = event.currentTarget.value;
+
+    setInfo((prevInfo) => {
+      // 업데이트된 값을 가진 새로운 배열 생성
+      const tempClicked = prevInfo.dayOff.map((day, idx) => {
+        if (idx === index) {
+          return day === '' ? DAYS[dayValue as keyof typeof DAYS] : '';
+        }
+        return day;
+      });
+
+      return {
+        ...prevInfo,
+        dayOff: tempClicked,
+      };
+    });
```

Comment:
```suggestion
    const tempClicked = info.dayOff.map((day, idx) => {
      if (idx === index) {
        return day === '' ? DAYS[dayValue as keyof typeof DAYS] : '';
      }
      return day;
    });

    handleInputChange('dayOff', tempClicked);
```

이렇게 해주고 `useEffect` 훅 제거해주면 `info`에 변경사항이 잘 저장되는 거 같습니다 !
이게 `console`을 찍어보면 `useState` 훅이 비동기 함수라 한 차례 늦게 반영되는 것처럼 보이는데 다른 액션을 통해서 `console` 두 번 찍어보면 두 번째 console에서 첫번째 때 했던 액션이 반영되는 걸 볼 수 있어요 ! 이렇게 하면 useEffect 훅 안 쓰고 해결하 수 있을 것 같습니다.

이렇게 함수 변경하면 `handleInputChange`를 상위로 넘겨주면 되겠죠 ?

Reply:
이 부분은 다른 브랜치에서 조금 대거 건드린 부분이 있어서! 요거는 마지막 api 연결과 함께 수정해두도록 하겠습니다!!